src/aatoolbox/datasources/usgs/usgs_ndvi.py

```diff
@@ -0,0 +1,924 @@
+from urllib.error import HTTPError
+from urllib.request import urlopen
+from zipfile import ZipFile
+
+import geopandas as gpd
+import pandas as pd
+import regex as re
+import rioxarray  # noqa: F401
+import xarray as xr
+from typing_extensions import Literal
+
+import aatoolbox.utils.raster  # noqa: F401
+from aatoolbox.config.countryconfig import CountryConfig
+from aatoolbox.datasources.datasource import DataSource
+from aatoolbox.utils._dates import (
+    _compare_dekads_gt,
+    _compare_dekads_lt,
+    _date_to_dekad,
+    _dekad_to_date,
+    _expand_dekads,
+)
+
+# TODO: use check_file_existence once the fix is merged
+# that allows it to work on class methods (PR is open)
+# from aatoolbox.utils.io import check_file_existence
+
+logger = logging.getLogger(__name__)
+
+_VALID_AREAS = Literal[
+    "north_africa",
+    "east_africa",
+    "southern_africa",
+    "west_africa",
+    "central_asia",
+    "yemen",
+    "central_america",
+    "hispaniola",
+]
+_VALID_TYPES = Literal[
+    "smoothed", "percent_median", "median_anomaly", "difference"
+]
+
+
+class _UsgsNdvi(DataSource):
+    """Base class to retrieve USGS NDVI data.
+
+    Parameters
+    ----------
+    country_config : CountryConfig
+        Country configuration
+    coverage_area : str
+        Area of coverage
+    data_type : str
+        Data type
+    start_date : Union[date, str, Tuple[int], None]
+        Start date. Can be passed as a ``datetime.date``
+        object and the relevant dekad will be determined,
+        as a date string in ISO8601 format, or as a
+        year-dekad tuple, i.e. (2020, 1).
+    end_date : Union[date, str, Tuple[int], None]
+        End date. Can be passed as a ``datetime.date``
+        object and the relevant dekad will be determined,
+        as a date string in ISO8601 format, or as a
+        year-dekad tuple, i.e. (2020, 1).
+    processed_file_suffix : str
+        Suffix for processed file.
+    """
+
+    def __init__(
+        self,
+        country_config: CountryConfig,
+        coverage_area: _VALID_AREAS,
+        data_type: _VALID_TYPES,
+        start_date: Union[date, str, Tuple[int], None] = None,
+        end_date: Union[date, str, Tuple[int], None] = None,
+        processed_file_suffix: str = "",
+    ):
+        super().__init__(
+            country_config=country_config,
+            datasource_base_dir="usgs_ndvi",
+            is_public=True,
+            is_global_raw=True,
+        )
+        self.coverage_area = coverage_area
+
+        # set data type directly in init because should not be
+        # changed in an instance
+        valid_types = _VALID_TYPES.__args__  # type: ignore
+        if data_type not in valid_types:
+            raise ValueError(
+                "`data_type` is not a valid string value. "
+                f"It must be one of {*valid_types,}"
+            )
+        self._data_type = data_type
+
+        # url code and file suffix
+        loc_data_types = {
```

Comment:
To me it makes more sense to not have a dictionary for this, but rather pass all the parameters (i.e. `data_type_url`, `data_type_suffix` ) directly to `__init__`. The reason for this is if you add another data type, then you would need to update the dictionary that's buried in the base class __init__. If you make them all input parameters, you would know right away that you need to provide this info. Hope that makes sense.

Reply:
That's very clear, thanks!